hscontrol/grpcv1.go

```diff
@@ -652,6 +654,64 @@ func (api headscaleV1APIServer) DeleteApiKey(
 	return &v1.DeleteApiKeyResponse{}, nil
 }
 
+// GetACL returns the current ACL.
+func (api headscaleV1APIServer) GetACL(
+	_ context.Context,
+	_ *v1.GetACLRequest,
+) (*v1.GetACLResponse, error) {
+	acl, err := api.h.db.GetACL()
+	if err != nil {
+		if db.IsNotFoundError(err) {
+			return nil, types.ErrACLPolicyNotFound
+		}
+		return nil, err
+	}
+
+	return &v1.GetACLResponse{
+		Policy: acl.Proto().GetPolicy(),
+	}, nil
+}
+
+// SetACL inserts or updates the ACL.
+func (api headscaleV1APIServer) SetACL(
+	_ context.Context,
+	request *v1.SetACLRequest,
+) (*v1.SetACLResponse, error) {
+	if api.h.cfg.ACL.PolicyMode != types.ACLPolicyModeDB {
+		return nil, types.ErrACLPolicyUpdateIsDisabled
+	}
+
+	polBytes, err := request.GetPolicy().MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+
+	a, err := policy.LoadACLPolicyFromBytes(polBytes, "json")
+	if err != nil {
+		return nil, errors.Wrap(err, types.ErrInvalidACLPolicyFormat.Error())
+	}
+
+	api.h.ACLPolicy = a
```

Comment:
Should this line be moved down to after the ACL is successfully persisted?

Reply:
Yes. I've moved it after we set the ACL in the database.